### PR TITLE
Add declarative installation

### PR DESCRIPTION
**Pull request by balamurugana**

### CODE REVIEW

File: docs/tools/install.sh [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+    fi
+
+    echo "* Generate DirectPV manifests to be saved at directpv.yaml"
+    if [[ "${legacy}" -eq 0 ]]; then
+        run_plugin_cmd install --declarative "${directpv_args[@]}" > directpv.yaml
+    else
+        run_plugin_cmd install --legacy --declarative "${directpv_args[@]}" > directpv.yaml
+    fi
+
+    if [ "${apply_flag}" -eq 1 ]; then
```
Comment (Praveenrajmani) [inline docs/tools/install.sh:201]:
```suggestion
    if [ "${apply_flag}" -eq 0 ]; then
```

Currently, it applies even if `apply` flag is not provided

File: docs/tools/install.sh [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+
+EXAMPLES:
+  # Generate DirectPV manifests.
+  $ install.sh
+
+  # Generate DirectPV manifests with private registry.
+  $ install.sh --registry private-registry.io --org org-name
+
+  # Generate DirectPV manifests with node-selector.
+  $ install.sh --node-selector node-label-key=node-label-value
```
Comment (Praveenrajmani) [inline docs/tools/install.sh:145]: Is there a way to test local dev images? can we also add `--image`?

File: pkg/installer/args.go [hunk 3]
```diff
@@ -34,40 +37,37 @@ const (
...
+	Tolerations      []corev1.Toleration
+	SeccompProfile   string
+	AppArmorProfile  string
+	Quiet            bool
+	KubeVersion      *version.Version
+	Legacy           bool
+	ObjectWriter     io.Writer
+	DryRun           bool
+	Declarative      bool
+	ObjectConverter  func(runtime.Object) ([]byte, error)
```
Comment (Praveenrajmani) [inline pkg/installer/args.go:56]: can we call it as Marshaller/ObjectMarshaller?

File: docs/tools/install.sh (hunk 2 above)
Comment (balamurugana) [reply docs/tools/install.sh:145]: You can pass whatever `kubectl-directpv install` command has. These examples are to indicate how to use `install` command's arguments.

File: docs/tools/install.sh (hunk 1 above)
Comment (balamurugana) [reply docs/tools/install.sh]: done

File: pkg/installer/args.go (hunk 3 above)
Comment (balamurugana) [reply pkg/installer/args.go]: done

File: docs/tools/install.sh (hunk 2 above)
Comment (Praveenrajmani) [reply docs/tools/install.sh:145]: this is not an issue also, we can edit the image manually in the manifest. just that we can't use the script

**Review by Praveenrajmani — APPROVED** [review]

File: docs/tools/install.sh (hunk 2 above)
Comment (balamurugana) [reply docs/tools/install.sh:145]: Your expectation is wrong. The image you are expecting has to be in the registry or in the local registry. It is a bad idea to modify the manifests here.

Comment (Praveenrajmani) [reply docs/tools/install.sh:145]: I meant we can do `kubectl directpv install --declarative > manifest.yaml` and then editing the `image` field to our dev image.

I see a point that `dev` images will not be of real use here. So, there is not need to use this script for development.

**Pull request merged**